### PR TITLE
Issue #642 Expo 33 upgrade and package updates

### PR DESCRIPTION
**Pull request by bensinjin**
After the upgrade I've ensured the app works as expected, Tomy and I also did a test to ensure we were able to properly build both iOS and Android binaries using `expo bi` and `expo ba` which was successful. We have upgraded the location library so I would like to test this once the staging build is available. According to this comment: https://github.com/expo/expo/issues/4374#issuecomment-509636802 the builders (on Expo servers) should now be producing 64 bit version of the app although if this is not the case we are told it will for sure be happening once version 34 of Expo is out.

I realize the ACs for this were to create stage releases but I see there being a bit of a blocker in terms of lack of proper documentation. Please point me to the documentation to ensure I have all translations, proper server configurations, rules around tagging releases etc... If this isn't readily available or perhaps a work in progress @tomy-pg should be able to take it from here as he's been more recently in charge of creating stage releases.

### CODE REVIEW

**Comment by ian-lin-PG** [comment]
I recommend upgrading the @types react and @types react-native version as well. Yarn is currently installing outdated type definitions. Reference: npm page for [react](https://www.npmjs.com/package/@types/react) and [react-native](https://www.npmjs.com/package/@types/react-native).

image
image
image
image

**Comment by bensinjin** [comment]
Thx @ian-lin-PG - done!

**Comment by ian-lin-PG** [comment]
I am able to use the feature I need to fix #525, I got no other issues from my testing.

**Comment by bensinjin** [comment]
Cheers @ian-lin-PG, feel free to mark this as "approved" by you (in the review tab) if you feel it's ready.

File: package.json [hunk 1]
```diff
@@ -9,11 +9,11 @@
         "@lingui/react": "2.7.3",
         "@types/expo": "^32.0.0",
         "@types/history": "^4.7.0",
-        "@types/jest": "^23.3.2",
+        "@types/jest": "^24.0.2",
         "@types/query-string": "^6.1.0",
         "@types/ramda": "^0.25.38",
-        "@types/react": "^16.0.40",
-        "@types/react-native": "^0.52.20",
+        "@types/react": "^16.8.23",
```
[the remark this answers is not in the document]
Comment (bensinjin) [reply package.json]: We do our best to keep our types definition as up to date as we can but it's not nearly as important as all of our other dependencies. I'm hoping we'll find as time goes on that these types will start being maintained in only one location, the main library (I've seen this for some), so we can start removing these deps.

[the remark this answers is not in the document]
Comment (rasmus-storjohann-PG) [reply package.json]: I don't have a very good sense of how the @types are manages for each important library, I can guess that it varies from project to project. Updating to the most recent available is probably always going to be the right course of action, unless we're running with a very old version of a library for some reason.

**Pull request merged**